File: v03/zad2/Valjak.cpp
#include "Valjak.hpp"

// Podrazumevani konstruktor, poziva konstruktore za M i B
Valjak::Valjak(double r, double h) : B(r), M(2 * r * M_PI, h) { }

// Vraća poluprečnik baze valjka
double Valjak::getR() const {
    return B.getR();
}

// Vraća visinu valjka
double Valjak::getH() const {
    return M.getB();
}

// Vraća površinu valjka
double Valjak::getP() const {
    return 2 * B.getP() + M.getP();
}

// Vraća zapreminu valjka
double Valjak::getV() const {
    return B.getP() * M.getB();
}

File: v10/zad2/Valjak.java
package zad2;

public class Valjak {
	// Polja
	private Krug b;
	private Pravougaonik m;
	
	// Konstruktor sa parametrima
	public Valjak(double r, double h) {
		b = new Krug(r);
		m = new Pravougaonik(b.obim(), h);
	}
	
	// Geteri
	public double getR() {
		return b.getR();
	}

	public double getH() {
		return m.getB();
	}
	
	// Metoda koja vraća površinu
	public double povrsina() {
		return 2 * b.povrsina() + m.povrsina();
	}
	
	// Metoda koja vraća zapreminu
	public double zapremina() {
		return b.povrsina() * getH();
	}
	
}
